src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -257,6 +261,7 @@ public String getVirtualMachineSize() {
         return virtualMachineSize;
     }
 
+    public String getStorageAccountType() { return storageAccountType; }
```

Comment:
when upgrading from a previous version storageAccountType will be null. You should add a check for this here and return Constants.STORAGE_ACCOUNT_TYPE_STANDARD when that happens.